chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -353,14 +355,20 @@ impl Ledger {
 
         let epoch = new_ledger.date.epoch + 1;
 
-        let expected_epoch_reward =
-            rewards::rewards_contribution_calculation(epoch, &ledger_params.reward_params);
+        // changed to scale rewards calculation
+        let expected_epoch_reward = rewards::rewards_contribution_calculation_scaled(
+            epoch,
+            &ledger_params.reward_params,
+            // Pass total stake for all pools from distribution -- KH
+            distribution.total_stake(),
+        );
 
         let mut total_reward = new_ledger.pots.draw_reward(expected_epoch_reward);
 
+        // Changed from true to false -- rewards *must* accrue to the treasury in the testnet.   KH
         // Move fees in the rewarding pots for distribution or depending on settings
         // to the treasury directly
-        if true {
+        if false {
```

Comment:
Not sure I understand this, since there is an if false, which is not doing the right thing (as agreed for the testnet).  We cannot move fees to rewards.  It is a vulnerability.

Reply:
this is a library, which is not about testnet or mainnet. we want this configurable and/or generics so that we can conduct tests/benchmarks of many possible scenario